man/adoc/bpftrace.adoc

```diff
@@ -1846,6 +1850,39 @@ the path will be clamped by `size` otherwise `BPFTRACE_MAX_STRLEN` is used.
 
 This function can only be used by functions that are allowed to, these functions are contained in the `btf_allowlist_d_path` set in the kernel.
 
+[#functions-percpu-kaddr]
+=== percpu_kaddr
+
+.variants
+* `void *percpu_kaddr(const string name)`
+* `void *percpu_kaddr(const string name, int cpu)`
+
+*sync*
+
+Get the address of the percpu kernel symbol `name` for CPU `cpu`. When `cpu` is
+omitted, the current CPU is used.
+
+----
+interval:s:1 {
+  $proc_cnt = kaddr("process_counts");
+  printf("% processes are running on CPU %d\n", *$proc_cnt, cpu);
+}
+----
+
+The function may return NULL if `cpu` is higher than the number of available
+CPUs. Therefore, it is necessary to perform a NULL-check on the result when
+accessing fields of the pointed structure, otherwise the BPF program will be
+rejected.
```

Comment:
What happens if the kernel symbol is invalid? e.g. `percpu_kaddr("potato")`

Reply:
Then we fail in the semantic analyser, see one of the unit tests.